### PR TITLE
Make all configs overrideable by environment variables

### PR DESCRIPTION
**Pull request by TheConnMan**
I plan on using Destalinator in a vanilla Docker container so having all configs as environment variables is a must. All `configuration.yaml` configs still work.

### CODE REVIEW

**Comment by zeckalpha** [comment]
This looks great! We talked about making the destalinator env bars all prefixed with DESTALINATOR_, what do you think of that?

–
Kyle Marek-Spartz

**Comment by TheConnMan** [comment]
Doesn't matter either way for a Docker container because there wouldn't be an env var name collisions. For running natively it makes sense to prefix. All you'd have to do is prepend `DESTALINATOR_` when checking for the env var inside the confg class.

File: config.py [hunk 1]
```diff
@@ -20,7 +20,13 @@ def __getattr__(self, attrname):
             warnings.warn("The `slack_name` key in %s is deprecated in favor of the `SLACK_NAME` environment variable" %
                           self.config_fname, DeprecationWarning)
 
-        return self.config[attrname]
+        if os.getenv(attrname.upper()) is not None:
+            return os.getenv(attrname.upper()).split(',') if ',' in os.getenv(attrname.upper()) else os.getenv(attrname.upper())
+
+        try:
```
Comment (zeckalpha) [inline config.py]: What was the exception here? KeyError? If so, `return self.config.get(attrname, '')` is a little cleaner.

File: configuration.yaml [hunk 2]
```diff
@@ -1,24 +1,11 @@
...
 # but no changes or messages should affect a live Slack).
-destalinator_activated_env_varname: DESTALINATOR_ACTIVATED
-
-# Name of environment variable determining whether output debug mode is enabled
-output_debug_env_varname: DESTALINATOR_SLACK_VERBOSE
-
-# Name of environment variable for the earliest date to archive stale channels.
-# If this is set, it should be of the form "yyyy-mm-dd" (e.g. "2017-02-19").
-earliest_archive_date_env_varname: EARLIEST_ARCHIVE_DATE
+destalinator_activated: false
```
Comment (zeckalpha) [inline configuration.yaml]: If we auto prefix env vars with DESTALINATOR_, this one could just be `activated: false`

File: README.md [hunk 3]
```diff
@@ -71,7 +73,7 @@ After saving, you can copy the OAuth Access Token value from the top of the same
 
 Destalinator can be chatty and make potentially big changes to a Slack team (by warning or archiving a large amount of channels), especially when first installed.
 
-To minimize the risk of making a mistake, Destalinator will run in a dry-run mode unless the `DESTALINATOR_ACTIVATED` environment variable exists. Set it to any non-empty value and Destalinator is "active." If you want to remain in dry-run mode, ensure this variable is unset/does not exist.
+To minimize the risk of making a mistake, Destalinator will run in a dry-run mode unless the `DESTALINATOR_ACTIVATED` environment variable exists. Set it to true and Destalinator is "active." If you want to remain in dry-run mode, ensure this variable is unset/does not exist.
```
Comment (zeckalpha) [inline README.md]: Oh I see, yes what you had is correct re: false.

File: configuration.yaml (hunk 2 above)
Comment (TheConnMan) [inline configuration.yaml]: Agreed, want me to throw in the `DESTALINATOR_` prefix in this PR? I couldn't tell if you talking about it internally meant you were ready to go forward with it. If so, I'll change the getter for this to be just `activated` as you said.

Comment (zeckalpha) [inline configuration.yaml]: That would be helpful! I would worry about changing it out from under you (or whoever else) later.

**Comment by TheConnMan** [comment]
There were a few env var only references throughout the project (logging level, test schedule) which I didn't touch as they don't have config properties to go with them. I'm all for making those config-able and piping them though the consolidated config.

File: config.py (hunk 1 above)
Comment (TheConnMan) [inline config.py]: I think I tried 4 different ways to access attributes (yay Python) which all threw accessor exceptions when you pulled an attribute which didn't exist. What you've suggested works perfectly for what I was trying to achieve with this try/catch. I'll update with those changes.

**Comment by TheConnMan** [comment]
I could use some additional testing, these changes ended up affecting a lot of code.

**Comment by zeckalpha** [comment]
I see one bug in the flagger, but otherwise I'll merge this to my work slack and try it out next week.

–
Kyle Marek-Spartz

**Comment by TheConnMan** [comment]
Awesome, thank you. Looking forward to rolling this out in my own work Slack.

**Comment by TheConnMan** [comment]
@zeckalpha How did the testing go last week?

**Pull request merged**